netbox-discovery-quickstart/6_start_device_discovery.sh

```diff
@@ -0,0 +1,81 @@
+#!/bin/bash
+set -euo pipefail
+
+# Check if all required environment variables are set
+REQUIRED_VARS=("MY_EXTERNAL_IP" "DOCKER_SUBNET" "NETBOX_PORT" "DIODE_API_KEY")
+
+for var in "${REQUIRED_VARS[@]}"; do
+  if [ -z "${!var:-}" ]; then
+    echo "Error: Required environment variable '$var' is not set."
+    exit 0
+  fi
+done
+
+WORKING_DIR="device_discovery"
+
+# Remove config directory if it exists
+sudo rm -fr ${WORKING_DIR}
+
+# Recreate it and pushd in
+mkdir ${WORKING_DIR}
+pushd ${WORKING_DIR}
+
+echo
+echo "--- Writing agent config ---"
+echo
+
+cat <<EOF > agent.yaml
+orb:
+  config_manager: 
+    active: local
+  backends:
+    device_discovery:
+    common:
+      diode:
+        target: grpc://${MY_EXTERNAL_IP}:8080/diode
+        api_key: ${DIODE_API_KEY}
+        agent_name: agent1
+  policies:
+    device_discovery:
+      discovery_1:
+        config:
+          schedule: "* * * * *"
+          defaults:
+            site: New York NY
+        scope:
+          - driver: srl
+            hostname: 172.24.0.100
+            username: admin
+            password: NokiaSrl1!
+            optional_args:
+               insecure: True
+          - driver: srl
+            hostname: 172.24.0.101
+            username: admin
+            password: NokiaSrl1!
```

Comment:
you can pass password as env variable as well to be more 'secure'

Reply:
Ok that's a good idea. I'll add it.